PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/model/PhylogeneticTreeNode.java

```diff
@@ -1,109 +1,240 @@
 package nl.tudelft.pl2016gr2.gui.model;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode;
+
 import java.util.ArrayList;
-import java.util.Objects;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 /**
- * This class is an adapter class which maps the methods of the
- * net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode class to the interface which is needed by
- * our application.
+ * This class is a storage class for a phylogenetic tree node. It retrieves and stores the needed
+ * values of the newick parser {@link TreeNode}.
  *
  * @author Faris
  */
-public class PhylogeneticTreeNode implements IPhylogeneticTreeNode {
+public class PhylogeneticTreeNode implements IPhylogeneticTreeNode, Iterable<PhylogeneticTreeNode> {
```

Comment:
Is the `Iterable<>` interface not a property of the `TreeNode` _interface_? Unless it is only specific to this implementation, you should do `IPhylogeneticTreeNode extends Iterable<IPhylogeneticTreeNode>` instead, to enforce all implementations to be iterable. Note that the iterable now is over the _interface_ rather than the implementation (mind the `I`).